src/app/shared/example-viewer/example-viewer.ts

```diff
@@ -19,15 +19,23 @@ export class ExampleViewer {
   /** Data for the currently selected example. */
   exampleData: LiveExample;
 
-  /** Whether the source for the example is being displayed. */
-  showSource = false;
-
   /** Component type for the current example. */
   _exampleComponentType: Type<any>|null = null;
 
   /** Module factory that declares the example component. */
   _exampleModuleFactory: NgModuleFactory<any>|null = null;
 
+  /** View of the example component. */
+  view: 'compact' | 'full' | 'collapsed' = 'compact';
+
+  /** Initial view of the example component. */
+  @Input() set initialView(initialView: 'compact' | 'full' | 'collapsed') {
+    this.view = initialView;
```

Comment:
You might want to have this just in the `ngOnInit`. I feel like there's something about Ivy that'll cause this to keep getting assigned to the same value every time.

Yea, I just checked the generated code for `ComponentExamples`:
```
function ComponentExamples_ng_container_0_example_viewer_3_Template(rf, ctx) { if (rf & 1) {
    _angular_core__WEBPACK_IMPORTED_MODULE_1__["ɵɵelement"](0, "example-viewer", 3);
} if (rf & 2) {
    const example_r62 = ctx.$implicit;
    _angular_core__WEBPACK_IMPORTED_MODULE_1__["ɵɵproperty"]("example", example_r62)("showCompactToggle", false)("initialView", "collapsed");
} }
```

This means that every time change detection runs on that component (`if (rf & 2) {...}`), the property is written back to 'collapsed'. So if you had changed the `view` value internally, it's going to get changed back.